### ISSUE TITLE
thousandSeparator & decimalScale conflict?

### ISSUE THREAD

**Issue by locpd** [s-yadav/react-number-format#178]
If I use thousandSeparator together with setting decimalScale to 0, I can only enter up to 4 digits:

`<NumberFormat thousandSeparator={true} decimalScale={0} />`

Entering the 5th digit will remove the last 3 digits, leaving only the first digit left.

**Comment by em-ka** [comment]
Hi, I'm experiencing a similar issue. I've prepared an example:

[![Edit 23z8j5p5zr](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/23z8j5p5zr)

**Comment by em-ka** [comment]
I guess the problem lies [here on the line 616](https://github.com/s-yadav/react-number-format/blob/master/src/number_format.js#L616):

```js
/*
* get the valid numerical values only before the decimal
* separator when decimal scale is 0, issue #145
*/

if(decimalScale === 0 && !format) {
  inputValue = (inputValue.match(/^\d*/g) || []).join('');
}
```

**Comment by ad1992** [comment]
This was my commit. I will look into this.

**Comment by s-yadav** [comment]
Fixed in 3.3.2
